be/src/exec/pipeline/project_operator.cpp

```diff
@@ -87,6 +89,38 @@ Status ProjectOperatorFactory::prepare(RuntimeState* state) {
     return Status::OK();
 }
 
+// TODO: Need a better CPU cost estimation method
+static size_t expression_costs(Expr* expr) {
+    if (expr->is_slotref()) {
+        return 0;
+    }
+    // Default cost is data size
+    size_t costs = 0;
+    for (auto child : expr->children()) {
+        costs += expression_costs(child);
+    }
+
+    size_t expr_cost = expr->type().get_slot_size();
+    if (expr->node_type() == TExprNodeType::BINARY_PRED) {
+        if (expr->op() == TExprOpcode::MULTIPLY || expr->op() == TExprOpcode::DIVIDE) {
+            return costs + expr_cost * expr_cost;
+        }
+    }
+
+    return costs + expr_cost;
+}
+
+size_t ProjectOperatorFactory::cpu_costs() {
```

Comment:
should do this work in FE, and pass a flag to BE. because FE has statistics, it could do cost estimation more better.